### ISSUE TITLE
Error when building docker image

### ISSUE THREAD

**Issue by jbomaileva** [OmniDB/OmniDB#674]
Hi !

I have this error when I'm trying to build the docker image :

```Step 5/8 : RUN wget -q https://omnidb.org/dist/2.11.0/omnidb-server_2.11.0-debian-amd64.deb  && dpkg -i /app/omnidb-server_2.11.0-debian-amd64.deb  && rm -rf omnidb-server_2.11.0-debian-amd64.deb
 ---> Running in 3cb01c21cd58
Selecting previously unselected package omnidb-server.
(Reading database ... 6758 files and directories currently installed.)
Preparing to unpack .../omnidb-server_2.11.0-debian-amd64.deb ...
Unpacking omnidb-server (2.11.0) ...
Setting up omnidb-server (2.11.0) ...
/var/lib/dpkg/info/omnidb-server.postinst: line 2: systemctl: command not found
/var/lib/dpkg/info/omnidb-server.postinst: line 3: systemctl: command not found
/var/lib/dpkg/info/omnidb-server.postinst: line 4: systemctl: command not found
dpkg: error processing package omnidb-server (--install):
 subprocess installed post-installation script returned error exit status 127
Errors were encountered while processing:
 omnidb-server
The command '/bin/sh -c wget -q https://omnidb.org/dist/2.11.0/omnidb-server_2.11.0-debian-amd64.deb  && dpkg -i /app/omnidb-server_2.11.0-debian-amd64.deb  && rm -rf omnidb-server_2.11.0-debian-amd64.deb' returned a non-zero code: 1```

**Comment by wind39** [comment]
Hi @jbomaileva ,

Thanks for reaching out. OmniDB 2.11 server packages now add a `systemctl` service, and in OmniDB `Dockerfile` we use the `debian:stable-slim` container (which does not have the `systemctl` command).

Just changing the container will fix this.

Regards,

William Ivanski

**Comment by eembsen** [comment]
Hi @wind39 
Can you please show me how? I am not able to build the image.

I guess you mean "Changing the base image will fix this" and I did that, but to no avail. I changed the base image to debian:latest and ubuntu:latest, and tried installing systemctl using "apt-get install systemd". The docker build always ends with:

`
Selecting previously unselected package omnidb-server.
(Reading database ... 5073 files and directories currently installed.)
Preparing to unpack .../omnidb-server_2.11.0-debian-amd64.deb ...
Unpacking omnidb-server (2.11.0) ...
Setting up omnidb-server (2.11.0) ...
Failed to connect to bus: No such file or directory
Created symlink /etc/systemd/system/multi-user.target.wants/omnidb.service, pointing to /etc/systemd/system/omnidb.service.
Failed to connect to bus: No such file or directory
dpkg: error processing package omnidb-server (--install):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 omnidb-server
`

Then I created the following Dockerfile:

`
FROM fedora:latest

RUN yum update -y && yum install -y wget dpkg

ENV container docker

RUN (cd /lib/systemd/system/sysinit.target.wants/; for i in *; do [ $i == systemd-tmpfiles-setup.service ] || rm -f $i; done); \
rm -f /lib/systemd/system/multi-user.target.wants/*;\
rm -f /etc/systemd/system/*.wants/*;\
rm -f /lib/systemd/system/local-fs.target.wants/*; \
rm -f /lib/systemd/system/sockets.target.wants/*udev*; \
rm -f /lib/systemd/system/sockets.target.wants/*initctl*; \
rm -f /lib/systemd/system/basic.target.wants/*;\
rm -f /lib/systemd/system/anaconda.target.wants/*;

VOLUME [ "/sys/fs/cgroup" ]

CMD ["/usr/sbin/init"]

RUN mkdir /app
WORKDIR /app

RUN wget -q https://omnidb.org/dist/2.11.0/omnidb-server_2.11.0-debian-amd64.deb \
 && dpkg -i /app/omnidb-server_2.11.0-debian-amd64.deb \
 && rm -rf omnidb-server_2.11.0-debian-amd64.deb

EXPOSE 8000
EXPOSE 25482

#CMD ["omnidb-server"]
'

A docker build of this Dockerfile ends with:

`
Selecting previously unselected package omnidb-server.
(Reading database ... 0 files and directories currently installed.)
Preparing to unpack .../omnidb-server_2.11.0-debian-amd64.deb ...
Unpacking omnidb-server (2.11.0) ...
Setting up omnidb-server (2.11.0) ...
System has not been booted with systemd as init system (PID 1). Can't operate.
Created symlink /etc/systemd/system/multi-user.target.wants/omnidb.service → /etc/systemd/system/omnidb.service.
System has not been booted with systemd as init system (PID 1). Can't operate.
dpkg: error processing package omnidb-server (--install):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 omnidb-server
`

Were you able to build the image? If yes, then what did you do? I suspect that there is something in the systemctl service that wants a complete systemd service including dbus to be running. That is why I created my own Dockerfile based on fedora. But that does not work either.

Cheer,
Erwin

**Comment by thomasboussekey** [comment]
Hello @eembsen ,

I've worked on a docker omnidb image on the following project, initialted by Taivo Kasper:

https://github.com/taivokasper/docker-omnidb

It's using a Alpine 3.8 with Omnidb 2.10 or 2.11

Hope this helps,
Regards,
Thomas

**Comment by eembsen** [comment]
Hi Thomas,

Thank you. I will have a look at it.

Cheers,
Erwin

**Comment by wind39** [comment]
Closing this. Fixed by @sseide in PR #872 .
